ocean_provider/routes.py

```diff
@@ -273,6 +264,55 @@ def checkURL():
     )
 
 
+@services.route('/assetInfo', methods=['POST'])
```

Comment:
using uppercase/lowercase in HTTP paths could really go wrong. I would suggest always using lowercase when defining API paths.

Also, what's the difference between `/checkURL` & `/assetInfo` and why are they separate?

Reply:
checkUrl takes an URL (http:///www.example.com) and check if your selected provider is able to access it and gets file information (length & type)
/assetInfo takes a DID with some files (you don't have the URLs) and will send back the same interface, but with updated values.

Let's say that you published www.example.com as DID1. and you are updating the content, but in the DDO you still have the old length of the file. By calling /assetInfo, provider can decrypt the URLs , check if they are still valid, get the file info and then you can update your DDO with the new info

Reply:
Cool, got it. So if both return the same thing maybe the paths should more reflect what they do and be a bit grouped together:

- `/check-url`
- `/check-url-encrypted`

But what we are actually checking and returning is info about the file behind the URL. So named after what they return, and more REST like maybe:

- `/fileinfo`
- `/fileinfo/encrypted`

Reply:
the fileinfo endpoint seems more logical, but I would drop the `decrypt` part.

Reply:
I mean even better if both endpoints could be combined into one `/fileinfo` endpoint and depending if passed value is a URI or a DID different functionality happens but with the same result

Reply:
now thinking this may look better / easier to understand:
- `GET /fileinfo/<url>`
- `GET /assetfilesinfo/<did>`

Reply:
if we want to make it future-proof and support multiple files then `/fileinfo` should in both cases return array of objects, then there's also no further need to split it up

Reply:
Update:  let's have POST /fileinfo  with a payload { "url":"xx"}  or { "did":"xx"}

Reply:
I'm on it.